### PR TITLE
Rewrite Script Canvas section on Creating Custom Nodes

### PR DESCRIPTION
**Pull request by willihay**
## Change summary

- Rewrote topic on [Creating Custom Nodes](https://deploy-preview-1286--o3deorg.netlify.app/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/) with info on latest technique using nodeables, XML, and Jinja templates.
- Removed deprecated topics.
- Added new higher level topic on [Creating New Nodes](https://deploy-preview-1286--o3deorg.netlify.app/docs/user-guide/scripting/script-canvas/creating-new-nodes/) that provides an overview of the 4 ways you can create a new node, including from within the Editor and programmatically. Includes links to topics on each method.
- Added glossary entry for [nodeables](https://deploy-preview-1286--o3deorg.netlify.app/docs/user-guide/appendix/glossary/#nodeable) to Appendix.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+{{< note >}}
+No dependency on the Script Canvas Gem is required to create script bindings that reflect to the behavior context.
+{{< /note >}}
+
+Refer to the documentation on [Creating Script Canvas nodes from the behavior context](programmer-guide/behavior-context) in the Script Canvas Programmer Guide for more information.
+
+## Custom nodes
+
+For the most control and flexibility when creating new nodes, including the ability to add new functionality, you can use the highly customizable node implementation mechanism called _nodeables_. Nodeables use XML configuration, Jinja templates, and an automatic C++ code generation tool to create custom Script Canvas nodes.
```
Comment (carlitosan) [inline content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md:57]: I would add., "This allows programmers to write classes in C++, using an SDK that will always be compatible with the Script Canvas Editor and runtime. The SDK will guarantee that the desired interface of the C++ class will be a useable node in the Editor, and an useable object during execution."

**Review by carlitosan — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md (hunk 1 above)
Comment (willihay) [reply content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md]: Thanks! Let me know if my paraphrasing is accurate.

**Review by carlitosan — APPROVED** [review]

File: content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+{{< note >}}
+No dependency on the Script Canvas Gem is required to create script bindings that reflect to the behavior context.
+{{< /note >}}
+
+Refer to the documentation on [Creating Script Canvas nodes from the behavior context](programmer-guide/behavior-context) in the Script Canvas Programmer Guide for more information.
+
+## Custom nodes
+
+For the most control and flexibility when creating new nodes, including the ability to add new functionality, you can use the highly customizable node implementation mechanism called _nodeables_. Nodeables use XML configuration, Jinja templates, and an automatic C++ code generation tool to create custom Script Canvas nodes. Using this mechanism, programmers can write classes in C++ using an SDK that is guaranteed to generate a class interface that provides a useable node in the Script Canvas Editor, and a useable object at runtime.
```
Comment (paucom) [inline content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md:57]:
```suggestion
For the most control and flexibility when creating new nodes, including the ability to add new functionality, you can use _nodeables_, a highly customizable node implementation mechanism. To create custom Script Canvas nodes, nodeables use XML configuration, Jinja templates, and an automatic C++ code generation tool. Programmers can use nodeables to write classes in C++ using an SDK that is guaranteed to generate a class interface that provides both a usable node in Script Canvas Editor and a usable object at runtime.
```
* Re-ordered some sentences for flow.
* Changed "useable" to the more common spelling "usable".

File: content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+![Example of node created from nodeables](/images/user-guide/scripting/script-canvas/nodeable-node-example.png)
```
Comment (paucom) [inline content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md:59]:
```suggestion
![Example node created from nodeables](/images/user-guide/scripting/script-canvas/nodeable-node-example.png)
```

**Review by paucom — CHANGES_REQUESTED** [review]
Here's a partial edit pass. So far, I reviewed the changes in all files except for the "Creating Custom Nodes in Script Canvas" topic and the related images. I'll review those tomorrow.

File: content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+To expose new C++ classes, methods, constants, data types, and events to Script Canvas, you can create script bindings to the behavior context. This is the most common method used to create new nodes for O3DE Gems and components.
+
+![Example of node created from behavior context binding](/images/user-guide/scripting/script-canvas/behavior-context-node-example.png)
+
+{{< note >}}
+Script bindings are available to any O3DE scripting system that uses the behavior context, such as Lua.
+{{< /note >}}
+
+{{< note >}}
+No dependency on the Script Canvas Gem is required to create script bindings that reflect to the behavior context.
```
[the remark this answers is not in the document]
Comment (willihay) [reply content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md:50]: Suggestion updated with your second option. That captures the essence of my intentions.

**Review by lsemp3d — APPROVED** [review]

**Review by micronAMZN — APPROVED** [review]

**Review by paucom — APPROVED** [review]

**Pull request merged**